An attempt to fix our failing builds

jakule opened the pull request:
Add checkout out action to fix `Error: The process '/usr/bin/git' failed with exit code 128` as mentioned in 
https://github.com/dorny/paths-filter/issues/60

jakule committed:
An attempt to fix our failing builds

zmb3 commented on .github/workflows/aws-e2e-tests-non-root.yaml:
```diff
@@ -29,6 +29,7 @@ jobs:
      outputs:
        changed: ${{ steps.changes.outputs.changed }}
      steps:
+       - uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4.1.0
```
It looks like the checkout is only necessary on the `merge_group` trigger, not on `pull_request`. Can we filter this out with an `if`?

jakule committed:
Add merge_group condition to checkout step in workflows

This update adds a condition to the checkout step in various GitHub workflows to ensure it only runs when the event_name is "merge_group".

jakule committed:
Fix syntax

reedloden commented on .github/workflows/aws-e2e-tests-non-root.yaml:
```diff
@@ -29,6 +29,9 @@ jobs:
      steps:
+       - name: Checkout
+         if: ${{ github.event_name == 'merge_group' }}
+         uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4.1.0
```
```suggestion
         uses: actions/checkout@v4
```

reedloden commented on .github/workflows/check-devbox.yaml:
```diff
@@ -15,6 +15,9 @@ jobs:
      steps:
+       - name: Checkout
+         if: ${{ github.event_name == 'merge_group' }}
+         uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4.1.0
```
```suggestion
         uses: actions/checkout@v4
```

reedloden commented on .github/workflows/doc-tests.yaml:
```diff
@@ -15,6 +15,9 @@ jobs:
     steps:
+      - name: Checkout
+        if: ${{ github.event_name == 'merge_group' }}
+        uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4.1.0
```
```suggestion
         uses: actions/checkout@v4
```

reedloden commented on .github/workflows/integration-tests-non-root.yaml:
```diff
@@ -15,6 +15,9 @@ jobs:
     steps:
+      - name: Checkout
+        if: ${{ github.event_name == 'merge_group' }}
+        uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4.1.0
```
```suggestion
         uses: actions/checkout@v4
```

reedloden commented on .github/workflows/unit-tests-integrations.yaml:
```diff
@@ -15,6 +15,9 @@ jobs:
      steps:
+       - name: Checkout
+         if: ${{ github.event_name == 'merge_group' }}
+         uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4.1.0
```
```suggestion
         uses: actions/checkout@v4
```

reedloden reviewed (commented):
Please use `actions/checkout@v4` instead of specific commit hash. We permit rolling tags for GitHub-blessed actions under `actions/*`.

jakule committed:
Use v4 tag for checkout action instead of pinned commit

Co-authored-by: Reed Loden <reed@goteleport.com>
